### PR TITLE
[WIP] transpile without type check

### PR DESCRIPTION
**Pull request by bartlomieju**

### CODE REVIEW

File: cli/flags.rs [hunk 1]
```diff
@@ -1042,6 +1045,16 @@ fn ca_file_arg_parse(flags: &mut Flags, matches: &clap::ArgMatches) {
   flags.ca_file = matches.value_of("cert").map(ToOwned::to_owned);
 }
 
+fn no_type_check_arg<'a, 'b>() -> Arg<'a, 'b> {
+  Arg::with_name("no-type-check")
+    .long("no-type-check")
+    .help("Don't check TS types")
```
Comment (kitsonk) [inline cli/flags.rs:1051]:
```suggestion
    .help("Transpile to JavaScript without enforcing any TypeScript type checking.")
```

**Comment by bartlomieju** [comment]
@kitsonk your input would be appreciated - this PR adds option to keep TS compiler worker thread without terminating it after a single message

**Comment by bartlomieju** [comment]
> This will be quite incomplete, only transpiling a single module without doing dependency analysis inside the compiler. This will only transpile a single module, or keep spinning up the compiler snapshot for every module (likely the latter). 

Impeccable timing 😄  Actually I added option to keep compiler thread in the background - results are promising, for `std/` it's about 4-5s compared to >45s for type check

> Also, I think you need to consider cache invalidation... let's say you run once with `--no-type-check` and then you do drop it, what are the expected outcomes/results?

Well, it will actually run full type check using "classic" route, but output of transpiled modules **should be the same**.

**Comment by kitsonk** [comment]
When do you spin it down then?  Keeping it running indefinitely seems like a bad idea.

**Comment by bartlomieju** [comment]
> When do you spin it down then? Keeping it running indefinitely seems like a bad idea.

Currently never, I want to measure it more precisely, but I don't expect it to have great impact - there are 8 threads created by V8 anyway...

**Comment by bartlomieju** [comment]
Time it takes to run std/ tests:
- macOS - `35.670364019s` - suite `25277ms`
- ubuntu - `26.229788624s` - suite `20287ms`
- windows - `28.4952388s` - suite `20971ms`
- ubuntu debug - `114.533790058s` - suite `81459ms`

**Comment by kitsonk** [comment]
More worried about memory usage in keeping it running.. like if you had a webserver.

**Comment by bartlomieju** [comment]
Not fast enough, closing without merge

**Pull request closed without merging**